services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -262,19 +262,29 @@ open class ApplicationService(
      * Hanke.
      */
     @Transactional
-    open fun deleteWithOrphanGeneratedHankeRemoval(applicationId: Long, userId: String) =
-        with(getById(applicationId)) {
-            cancelAndDelete(this, userId)
-            if (hanke.generated && hanke.hakemukset.size == 1) {
-                logger.info {
-                    "Application $id was the only one of a generated Hanke, removing Hanke ${hanke.hankeTunnus}."
-                }
-                hankeRepository.delete(hanke)
-                val hankeDomain = domainFrom(hanke, geometryMapFrom(hanke))
-                hankeLoggingService.logDelete(hankeDomain, userId)
+    open fun deleteWithOrphanGeneratedHankeRemoval(
+        applicationId: Long,
+        userId: String
+    ): ApplicationDeletionResultDto {
```

Comment:
It doesn't feel great to have the service produce a specific DTO like this. But returning just a boolean would be pretty misleading as well. So I guess this is fine.

Reply:
I decided to use a dto because a boolean is not very self explanatory. Is the problem in the usage of a dto, or something else? If this function may or may not delete a hanke, it is definitely beneficial to know what did or did not happen.